sites/svelte.dev/src/lib/server/markdown/renderer.js

```diff
@@ -0,0 +1,753 @@
+import { modules } from '$lib/generated/type-info.js';
+import { createHash } from 'crypto';
+import fs from 'fs';
+import MagicString from 'magic-string';
+import { createShikiHighlighter, renderCodeToHTML, runTwoSlash } from 'shiki-twoslash';
+import ts from 'typescript';
+import { SHIKI_LANGUAGE_MAP, escape, normalizeSlugify, slugify, transform } from '../markdown';
+
+const METADATA_REGEX = /(?:<!---\s*([\w-]+):\s*(.*?)\s*--->|\/\/\/\s*([\w-]+):\s*(.*))\n/gm;
+
+const snippet_cache = new URL('../../../../node_modules/.snippets', import.meta.url).pathname;
+if (!fs.existsSync(snippet_cache)) {
+	fs.mkdirSync(snippet_cache, { recursive: true });
+}
+
+const type_regex = new RegExp(
+	`(import\\(&apos;svelte&apos;\\)\\.)?\\b(${modules
+		.flatMap((module) => module.types)
+		.map((type) => type.name)
+		.join('|')})\\b`,
+	'g'
+);
+
+const type_links = new Map();
+
+modules.forEach((module) => {
+	const slug = slugify(module.name);
+
+	module.types.forEach((type) => {
+		const link = `/docs/${slug}#type-${slugify(type.name)}`;
```

Comment:
Does this hold true for both Svelte and SvelteKit? I don't think it does, because all the types are on one page right now. I think we need some kind of configurability from the outside here.

Reply:
Probably yes. But I don't want to add that much configurability yet, before I move it to site-kit and test in both this site and kit. Maybe my assumptions will be wrong if I do em right now without real testing in Kit too